### ISSUE TITLE
fine-tuning to downstram task

### ISSUE THREAD

**Issue by zyangH1** [facebookresearch/dinov2#41]
Your work is so awesome. I am migrating vit-small to my task as a feature extractor, is this the right way to load pretrained-weight below. But it did not succeed in my task. But Dino worked well.

`# dinov2 model
model = vit_small(img_size = 224,block_chunks=0,mask = None,drop_path_rate=0.1,patch_size=14)
static_dict = torch.hub.load_state_dict_from_url('https://dl.fbaipublicfiles.com/dinov2/dinov2_vits14/dinov2_vits14_pretrain.pth')
interpolate_pos_embed(model,static_dict) # position adaption 

msg = model.load_state_dict(static_dict,strict=False)

print('miss keys: ',msg.missing_keys)
print('unpxpected keys: ',msg.unexpected_keys)`

**Comment by zyangH1** [comment]
I found the correct way to load the model in hubconf.py. I need to set init_value to 1.0 . I close this issue.

**Comment by TimDarcet** [comment]
Hi,
The model was at adapted to resolution 518 after pretraining, meaning it's made to take (518/14)**2+1=1370 tokens.
Your model is made to take as input res 224 images ((224/14)**2+1=257 tokens).
You should declare your model with `img_size=518`

Also, it's probably easier to just do
```
model = torch.hub.load('facebookresearch/dinov2', 'dinov2_vits14')
```
There should not be issues with this syntax

**Comment by Hexinx** [comment]
> Hi, The model was at adapted to resolution 518 after pretraining, meaning it's made to take (518/14)**2+1=1370 tokens. Your model is made to take as input res 224 images ((224/14)**2+1=257 tokens). You should declare your model with `img_size=518`
> 
> Also, it's probably easier to just do
> 
> ```
> model = torch.hub.load('facebookresearch/dinov2', 'dinov2_vits14')
> ```
> 
> There should not be issues with this syntax

Thank you and your excellent works. But why you set to 518? I'm confusing. Could you explain it? Very thanks.

**Comment by TimDarcet** [comment]
We simply wanted a higher resolution than pretraining (to achieve the effect shown in fig. 6), so any high resolution would have been suitable. 518 is just the multiple of 14 closest to 512, an often used resolution
